src/abstractdataframe/selection.jl

```diff
@@ -902,27 +906,38 @@ $TRANSFORMATION_COMMON_RULES
   column names should include the name of transformation functions or not.
 - `ungroup::Bool=true` : whether the return value of the operation on `gd` should be a data
   frame or a `GroupedDataFrame`.
+- `multithreaded::Bool=true` : whether transformations may be run in separate tasks which
+  can execute in parallel (possibly being applied to multiple rows or at the same time).
+  Whether or not tasks are actually spawned and their number are determined automatically.
+  Set to `false` if some transformations require serial execution or are not thread-safe.
 
 See [`select`](@ref) for examples.
 """
-select!(df::DataFrame, @nospecialize(args...); renamecols::Bool=true) =
+select!(df::DataFrame, @nospecialize(args...);
+        renamecols::Bool=true, multithreaded::Bool=true) =
     _replace_columns!(df, select(df, args..., copycols=false, renamecols=renamecols))
```

Comment:
here - and in other places where `select` is called internally we need to pass `multithreaded` argument to it.

Unfortunately it will be a bit of running search to locate all places where `select[!]`, `transform[!]`, `combine`, and `subset[!]` are called. It should not be many places but there are some.

Reply:
Good catch. I've checked again, I've found a few cases. In particular I added `multithreaded` to `unstack`. We could also add it to `describe` once we enable multithreading for `DataFrame` inputs -- I could add it now but it would never have any effect.

Reply:
> I could add it now but it would never have any effect.

Can you then please leave a "todo" comment in the code so that we do not forget to change it when we add multi-threading to `DataFrame` transformations?

Reply:
there are more places like this, e.g. `nonunique` etc. Maybe this TODO comment should be at some other place but I think we should add it.

Reply:
Maybe I can add tests instead, as they are less likely to go unnoticed if they break than a TODO.

Reply:
OK - great (but there are many such places - over 10 by my quick look, so as you prefer)

Reply:
Actually there are only two places where we pass to `select` an argument that is not guaranteed to be only column names: `describe` and `nonunique`. So I've added tests for both.